### PR TITLE
Reafctored the one click payment and fixed some bugs

### PR DESCRIPTION
**Pull request by cezarlz**

### CODE REVIEW

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php [hunk 1]
```diff
@@ -295,7 +295,7 @@ protected function save_user_meta_fields($order)
 	 */
 	public function process_payment($order_id)
 	{
-		if ( isset( $_POST['ebanx_billing_instalments'] ) ) {
+		if ( WC_EBANX_Request::has('ebanx_billing_instalments') ) {
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: Nice! 😄

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 2]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
 	protected function customer_has_ebanx_required_data() {
 		$card = current( array_filter( (array) array_filter( get_user_meta( $this->userId, '_ebanx_credit_card_token', true ) ), function ( $card ) {
-					return $card->token == WC_EBANX_Request::read('_ebanx_one_click_token');
-				} ) );
+			return $card->token == WC_EBANX_Request::read('ebanx-one-click-token');
+		} ) );
+
 		$names = $this->gateway->names;
 
 		$_POST['ebanx_token'] = $card->token;
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 3]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
 		$_POST['ebanx_masked_card_number'] = $card->masked_number;
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 4]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
 		$_POST['ebanx_brand'] = $card->brand;
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 5]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
-		$_POST['ebanx_billing_cvv'] = WC_EBANX_Request::read('_ebanx_one_click_cvv');
+		$_POST['ebanx_billing_cvv'] = WC_EBANX_Request::read('ebanx-one-click-cvv');
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 6]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
 		$_POST['ebanx_is_one_click'] = true;
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 7]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
-		$_POST['ebanx_billing_instalments'] = WC_EBANX_Request::read('_ebanx_one_click_installments');
+		$_POST['ebanx_billing_instalments'] = WC_EBANX_Request::read('ebanx-credit-card-installments');
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 8]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
 
 		$_POST[$names['ebanx_billing_brazil_document']] = get_user_meta( $this->userId, '_ebanx_billing_brazil_document', true );
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 9]
```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
...
 		$_POST[$names['ebanx_billing_brazil_birth_date']] = get_user_meta( $this->userId, '_ebanx_billing_brazil_birth_date', true );
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 10]
```diff
@@ -428,20 +292,14 @@ protected function customer_has_ebanx_required_data() {
 		$_POST['billing_city']      = $this->get_user_billing_address()['city'];
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 11]
```diff
@@ -428,20 +292,14 @@ protected function customer_has_ebanx_required_data() {
...
 		$_POST['billing_state']     = $this->get_user_billing_address()['state'];
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`

**Comment by cezarlz** [comment]
@gpressutto5 Well.. It was a not good idea translate to the `WC_EBANX_Request`... I replaced all methods, but the one click break down again, because the request_data and process_payment using yet the `$_POST`.

I think it's better stay this way, for now...

Thanks!

**Pull request merged**